include/ensmallen_bits/callbacks/early_stop_at_min_loss_on_validation.hpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file early_stop_at_min_loss_On_Validation.hpp
+ * @author Marcus Edel
+ * @author Omar Shrit
+ * 
+ * Implementation of the early stop at minimum loss callback function.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_CALLBACKS_EARLY_STOP_AT_MIN_LOSS_ON_VALIDATION_HPP
+#define ENSMALLEN_CALLBACKS_EARLY_STOP_AT_MIN_LOSS_ON_VALIDATION_HPP
+
+namespace ens {
+
+/**
+ * Early stopping to terminate the optimization process early if the loss stops
+ * decreasing.
+ */
+class EarlyStopAtMinLossOnValidation
```

Comment:
I guess instead of writing another callback class, we could also add another constructor to the existing  `EarlyStopAtMinLoss` class, let me know what you think, might avoid some code duplication.

Reply:
I agree I prefer one class with two constructors, the only reason for this prototype is our last discussion on the related issue. I will merge the two classes and push the code.

Reply:
Right, I guess that changed a bit because it's now part of ensmallen.